### DISCUSSION TITLE
Understanding why `requestBody.required` is expected as `true` in `OpenApiFactoryTest`

### DISCUSSION THREAD

**Discussion by monitaurus** [api-platform/core#6371] [Q&A] [answered]
Hi there.

I'm working on a PR related to this discussion https://github.com/api-platform/api-platform/discussions/2680, to make optional parameters in the OpenAPI `RequestBody` model.

My question doesn't evolve around my code, but I'm trying to understand a test case on the dummy `postDummyCollection` within `OpenApiFactoryTest` ([see here](https://github.com/api-platform/core/blob/e867d07f59b82d5f1bdca69e096ddf452dd7efc8/src/OpenApi/Tests/Factory/OpenApiFactoryTest.php#L173)). 

In the assertion part, it's expected for `postDummyCollection` `RequestBody`'s `required` to be `true` ([see here](https://github.com/api-platform/core/blob/e867d07f59b82d5f1bdca69e096ddf452dd7efc8/src/OpenApi/Tests/Factory/OpenApiFactoryTest.php#L608)).

But the dummy is built straight with the `$baseOperation` ([line 83](https://github.com/api-platform/core/blob/e867d07f59b82d5f1bdca69e096ddf452dd7efc8/src/OpenApi/Tests/Factory/OpenApiFactoryTest.php#L83)) which doesn't state anything about the `RequestBody`. In that case, I was thinking `required` should be `false`, as it's the default value in the constructor ([see constructor](https://github.com/api-platform/core/blob/e867d07f59b82d5f1bdca69e096ddf452dd7efc8/src/OpenApi/Model/RequestBody.php#L20)).

Does someone know what's the logical path involve for the `core` to generate a `required` `true` in this test case?

**Comment by soyuka** [comment, marked as answer]
On API Platform the body is required on every "write" operations.

**Reply by monitaurus** [reply]
It's totally make sens, thanks.
I changed my code to comply with this.

But while looking at the rest of the test cases, I stumble upon this dummy [postDummyCollectionWithRequestBody](https://github.com/api-platform/core/blob/e867d07f59b82d5f1bdca69e096ddf452dd7efc8/src/OpenApi/Tests/Factory/OpenApiFactoryTest.php#L183-L204), which define a `POST` operation, without the `required` parameter:
```php
'postDummyCollectionWithRequestBody' => (new Post())->withUriTemplate('/dummiesRequestBody')->withOperation($baseOperation)->withOpenapi(new OpenApiOperation(
    requestBody: new RequestBody(
        description: 'List of Ids',
        content: new \ArrayObject([
            'application/json' => [ ... ],
        ]),
    ),
)),
```

Based on current `Model\RequestBody` parameters ([see here](https://github.com/api-platform/core/blob/e867d07f59b82d5f1bdca69e096ddf452dd7efc8/src/OpenApi/Model/RequestBody.php#L20)), the default of `required` is `false`.
So the assert ends up testing `required` to be false ([see here](https://github.com/api-platform/core/blob/e867d07f59b82d5f1bdca69e096ddf452dd7efc8/src/OpenApi/Tests/Factory/OpenApiFactoryTest.php#L870)).

This assert seems logic to me given the actual code, but based on the rule you mentioned @soyuka _"the body is required on every "write" operations"_, this dummy seems to show a way to create an operation that contradicts it.

Is there another thingI didn't catch that explain this?

**Reply by soyuka** [reply]
open api allows this. it's probable that there's a bug in how API Platform describes the OpenAPI documentation in that regard.

**Reply by monitaurus** [reply]
Thanks for the clarification.
I'll make the PR comply with the current implementation of API Platform.
